uberfire-extensions/uberfire-metadata/uberfire-metadata-backends/uberfire-metadata-backend-infinispan/src/main/java/org/uberfire/ext/metadata/backend/infinispan/provider/InfinispanContext.java

```diff
@@ -0,0 +1,218 @@
+import org.uberfire.commons.lifecycle.Disposable;
+import org.uberfire.ext.metadata.backend.infinispan.exceptions.InfinispanException;
+import org.uberfire.ext.metadata.backend.infinispan.proto.KObjectMarshaller;
+import org.uberfire.ext.metadata.backend.infinispan.proto.schema.Schema;
+import org.uberfire.ext.metadata.backend.infinispan.proto.schema.SchemaGenerator;
+import org.uberfire.ext.metadata.backend.infinispan.utils.AttributesUtil;
+import org.uberfire.ext.metadata.model.KObject;
+
+import static java.util.stream.Collectors.toList;
+
+public class InfinispanContext implements Disposable {
+
+    private static final String PORT = "org.appformer.ext.metadata.infinispan.port";
+    private static final String HOST = "org.appformer.ext.metadata.infinispan.host";
+    private static final String USERNAME = "org.appformer.ext.metadata.infinispan.username";
+    private static final String PASSWORD = "org.appformer.ext.metadata.infinispan.password";
+
+    private static final String TYPES_CACHE = "types";
+    private static final String SCHEMAS_CACHE = "schemas";
+    private static final String PROTO_EXTENSION = ".proto";
+    private static final String SCHEMA_PROTO = "schema.proto";
+    private static final String ORG_KIE = "org.kie.";
+    private final RemoteCacheManager cacheManager;
+    private final KieProtostreamMarshaller marshaller = new KieProtostreamMarshaller();
+    private final SchemaGenerator schemaGenerator;
+
+    private final InfinispanConfiguration infinispanConfiguration;
+
+    private static InfinispanContext INSTANCE;
+
+    public static InfinispanContext getInstance() {
+        if (INSTANCE == null) {
+            Map<String, String> properties = new HashMap<String, String>() {{
+                put(HOST,
+                    System.getProperty(HOST,
+                                       "127.0.0.1"));
+                put(PORT,
+                    System.getProperty(PORT,
+                                       "11222"));
+                put(USERNAME,
+                    System.getProperty(USERNAME,
+                                       "user"));
+                put(PASSWORD,
+                    System.getProperty(PASSWORD,
+                                       "user"));
+            }};
+            INSTANCE = new InfinispanContext(properties);
+        }
+        return INSTANCE;
+    }
+
+    public static InfinispanContext getInstance(Map<String, String> properties) {
+        if (INSTANCE == null) {
+            INSTANCE = new InfinispanContext(properties);
+        }
+        return INSTANCE;
+    }
+
+    private InfinispanContext(Map<String, String> properties) {
+        this.infinispanConfiguration = new InfinispanConfiguration();
+        schemaGenerator = new SchemaGenerator();
+        ConfigurationBuilder builder = new ConfigurationBuilder();
+        builder.addServer()
+                .host(properties.get(HOST))
```

Comment:
In cases when user creates this class using `static InfinispanContext getInstance(Map<String, String> properties)` those properties might be empty. Shouln't we check the values and validate them (e.g. use the defaults if properties.get(HOST) is null or empty or fail fast in that case)?

Reply:
that method shouldn't be there, I forgot to remove it